db/error_handler.cc

```diff
@@ -412,6 +412,31 @@ void ErrorHandler::SetBGError(const Status& bg_status,
     recover_context_ = context;
     return;
   }
+  if (wal_related) {
+    assert(reason == BackgroundErrorReason::kWriteCallback ||
+           reason == BackgroundErrorReason::kMemTable ||
+           reason == BackgroundErrorReason::kFlush);
+  }
+  if (db_options_.manual_wal_flush && wal_related && bg_io_err.IsIOError()) {
+    // TODO: remove parameter `wal_related` once we can automatically recover
+    //  from WAL write failure.
+    // We should not try auto recover IOError from writing to WAL .
```

Comment:
nit: once we fix the bug in automatically recovery from WAL write failure. Before that, we should not try auto recovery IOError ....